src/Aspire.Hosting.Azure.AppService/AzureAppServiceInfrastructure.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Azure.AppService;
+
+internal sealed class AzureAppServiceInfrastructure(
+    ILogger<AzureAppServiceInfrastructure> logger,
+    IOptions<AzureProvisioningOptions> provisioningOptions,
+    DistributedApplicationExecutionContext executionContext) :
+    IDistributedApplicationLifecycleHook
+{
+    public async Task BeforeStartAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken = default)
+    {
+        if (!executionContext.IsPublishMode)
+        {
+            return;
+        }
+
+        var appServiceEnvironment = appModel.Resources.OfType<AzureAppServiceEnvironmentResource>().FirstOrDefault()
+            ?? throw new InvalidOperationException("AppServiceEnvironmentResource not found.");
+
+        var appServiceEnvironmentContext = new AzureAppServiceEnvironmentContext(
+            logger,
+            executionContext, 
+            appServiceEnvironment);
+
+        foreach (var resource in appModel.Resources)
+        {
+            // We only support project resources for now.
+            if (resource is not ProjectResource)
+            {
+                continue;
+            }
+
+            var website = await appServiceEnvironmentContext.CreateAppServiceAsync(resource, provisioningOptions.Value, cancellationToken).ConfigureAwait(false);
+
+#pragma warning disable ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+            resource.Annotations.Add(new DeploymentTargetAnnotation(website)
+            {
+                ContainerRegistry = appServiceEnvironment,
+                ComputeEnvironment = appServiceEnvironment
+            });
+#pragma warning restore ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates.
+        }
+
+        static void SetKnownParameterValue(AzureBicepResource r, string key, Func<AzureBicepResource, object> factory)
+        {
+            if (r.Parameters.TryGetValue(key, out var existingValue) && existingValue is null)
+            {
+                var value = factory(r);
+
+                r.Parameters[key] = value;
+            }
+        }
+
+        // Resolve the known parameters for the container app environment
+        foreach (var r in appModel.Resources.OfType<AzureBicepResource>())
```

Comment:
I'm going to remove this hack when we do https://github.com/dotnet/aspire/issues/8810. Generally, now that we have a more fully connected resource graph, a lot less need for these known parameters.

Reply:
Turns out I don't need to do this because we never need to work in this mode.